src/components/Personalization/createExecuteDecisions.js

```diff
@@ -23,9 +23,12 @@ const buildActions = decision => {
     scopeDetails: decision.scopeDetails
   };
 
-  return decision.items.map(item =>
-    assign({ type: DEFAULT_ACTION_TYPE }, item.data, { meta })
-  );
+  return decision.items.map(item => {
+    if (item.characteristics && item.characteristics.trackingLabel) {
+      meta.trackingLabel = item.characteristics.trackingLabel;
+    }
+    return assign({ type: DEFAULT_ACTION_TYPE }, item.data, { meta });
```

Comment:
If you want each item in the decision to have the exact same assigned tracking label, then this is fine. Otherwise, you have a bug.

Run this code and examine the result
```js
const DEFAULT_ACTION_TYPE = "default_action_type";

const assign = (...args) => Object.assign({}, ...args);

const buildActions = decision => {
  const meta = {
    id: decision.id,
    scope: decision.scope,
    scopeDetails: decision.scopeDetails
  };

  return decision.items.map(item => {
    if (item.characteristics && item.characteristics.trackingLabel) {
      meta.trackingLabel = item.characteristics.trackingLabel;
    }
    return assign({ type: DEFAULT_ACTION_TYPE }, item.data, { meta });
  });
};

const decision = {
  id: "decision-id",
  scope: "decision-scope",
  scopeDetails: "decision-scope-details",
  items: [
    {
      characteristics: {
        trackingLabel: "tracking-label-1"
      },
      data: { value: "lorem" }
    },
    {
      characteristics: {
        trackingLabel: "tracking-label-2"
      },
      data: { value: "ipsum" }
    },
    {
      characteristics: {
        trackingLabel: "tracking-label-3"
      },
      data: { value: "sit dolor" }
    }
  ]
};

console.log(buildActions(decision));
```
All the items have the `meta.trackingLabel` of the final `item` because of the shared `meta` object. Line 28 edits the reference and then line 30 adds a reference to final object. If you want to have each item have a different label, then you need to create a new `meta` object in each iteration.

Reply:
Good catch, thanks @carterworks !

Reply:
Fixed

Reply:
Nicely done.